src/consumer/consumerGroup.js

```diff
@@ -368,13 +368,34 @@ module.exports = class ConsumerGroup {
         )
 
         const leaders = keys(partitionsPerLeader)
+        const committedOffsets = this.offsetManager.committedOffsets()
```

Comment:
Is there a reason why we're using `committedOffsets` here, instead of just resolved? An invalid offset _could_ come  from attempting to resume from a committed offset, but also from a `consumer.seek`. Your great comment touches on how both are cleared, so I'm not sure whether there would be any actual difference in behaviour, but as future changes are made this subtle difference might be harder to spot while becoming more consequential.

Reply:
Initially I was actually using `resolvedOffsets`, but that didn't work because `OffsetManager.resolveOffsets` actually just sets the initialized consumer offsets in `committedOffsets`, not in `resolvedOffsets` (did someone mention that our naming is confusing...? 😅). When the consumer first boots, it doesn't actually have any resolved offsets, so the only source of offsets is the initialized offsets in `committedOffsets`.

Regarding the seek behavior, I would expect it to work the same way, no? Seek would commit (potentially invalid) offsets and then clear both `committedOffsets` and `resolvedOffsets` using `OffsetManager.clearOffsets`. In the fetch loop we'd get the consumer offsets from the brokers and from there on it's the same.

Maybe I'm missing something?

Reply:
> Regarding the seek behavior, I would expect it to work the same way, no? Seek would commit (potentially invalid) offsets 

Seeking shouldn't commit, only move the "playhead", see #395, so to rely on that behaviour is probably not the thing we want.

> Initially I was actually using resolvedOffsets, but that didn't work because OffsetManager.resolveOffsets actually just sets the initialized consumer offsets in committedOffsets, not in resolvedOffsets (did someone mention that our naming is confusing...? 😅).

I guess having to use `committedOffsets` is a symptom of there being an issue in there then. Conceptually, it's the `resolvedOffsets` (which I understand is the "next to consume offset" or  "playhead" for reading the log) that should always exist and the committed offset which is optional, as using Kafka for committing offsets is / should be totally optional (see #395). 

Since that seems like a  different issue, maybe it's an idea we create a separate issue for it and tag that in a comment. Being able to spot outside of the context of these changes that we conceptually want the resolved offsets rather than committed there might be a lot to ask from our future selves (or others) 😅.

Reply:
> Since that seems like a different issue, maybe it's an idea we create a separate issue for it and tag that in a comment.

That sounds like a good idea. I would prefer to do that kind of holistic refactoring in a PR that doesn't actually change any behavior, rather than squeezing it into a bugfix.

Reply:
Created the issue, trying to preserve the context of this conversation properly: #585. To help the audit suggested in there, I'd suggest linking that issue in a comment above where `committedOffsets()` is called.